### PR TITLE
Give space dragon ability to make it's sprite small

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

Allows space dragon to replace it's sprite with a carp sprite like the xeno queen can turn it's sprite into a drone. 

This means only the dragon sees itself as small. Others still see it as giant.

## Why It's Good For The Game

The space dragon sprite is pretty big, so you can't attack stuff directly north of you. This solves that.

## Changelog

:cl:
qol: space dragon can now make it's sprite smaller
/:cl:

### CODE REVIEW

File: code/datums/action.dm [hunk 1]
```diff
@@ -873,8 +873,6 @@
 
 /datum/action/small_sprite/megafauna
 	icon_icon = 'icons/mob/actions/actions_xeno.dmi'
-	button_icon_state = "smallqueen"
-	background_icon_state = "bg_alien"
```
Comment (SuperNovaa41) [inline code/datums/action.dm:877]: Why did you remove these?

Comment (ivanmixo) [reply code/datums/action.dm:877]: Because it's a subtype of `/datum/action/small_sprite`, which has these vars already defined as these. No need to do it twice.

Comment (SuperNovaa41) [reply code/datums/action.dm:877]: oh yes i didn't see that

**Comment by ivanmixo** [comment]
AaaaaaAAAAAAAAAAAAAAAA

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 2]
```diff
@@ -201,10 +206,20 @@
 		destroy_rifts()
 	..()
 	add_dragon_overlay()
+	rift.Remove(src)
+	QDEL_NULL(rift)
+	small_sprite.Remove(src)
+	UnregisterSignal(small_sprite, COMSIG_ACTION_TRIGGER)
+	QDEL_NULL(small_sprite)
```
Comment (MrMelbert) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:213]: you don't need to call `remove()`, `qdel`ing an action will do it for you

Comment (MrMelbert) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:213]: another question: why do you delete all the actions whenever they die, anyways?

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm (hunk 2 above)
Comment (ivanmixo) [reply code/modules/mob/living/simple_animal/hostile/space_dragon.dm:213]: I can't remember for the life of me tbh

**Comment by ivanmixo** [comment]
No
